### PR TITLE
executor: make the format of user and host fields in slow logs be compatible with MySQL

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: related to #17846 

Problem Summary: make the format of user and host fields in slow logs be compatible with MySQL

### What is changed and how it works?
The format of TiDB now is: `# User: {user_name}@{host_address}`
For example: `# User: root@127.0.0.1`

After this PR, it will be: `# User@Host: {user_name}[{user_name}] @ {host_name} [{host_address}]`, which is compatible with MySQL.
For example: `# User@Host: root[root] @ localhost [127.0.0.1]`

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test

### Release note 

- executor: make the format of user and host fields in slow logs be compatible with MySQL

### CODE REVIEW

**Comment by qw4990** [comment]
> LGTM

Thx @crazycs520 , PTAL @SunRunAway @XuHuaiyu

File: executor/slow_query.go [hunk 1]
```diff
@@ -373,13 +384,16 @@ func (st *slowQueryTuple) setFieldValue(tz *time.Location, field, value string,
 		}
 	case variable.SlowLogTxnStartTSStr:
 		st.txnStartTs, err = strconv.ParseUint(value, 10, 64)
-	case variable.SlowLogUserStr:
```
Comment (SunRunAway) [inline executor/slow_query.go:376]: Does this case need to be kept for the capability of old format?

Comment (qw4990) [reply executor/slow_query.go:376]: updated. PTAL.

**Pull request merged**